src/workerd/jsg/async-context.c++

```diff
@@ -29,11 +29,14 @@ AsyncContextFrame::AsyncContextFrame(Lock& js, StorageEntry storageEntry)
     // If current(js) returns nullptr, we assume we're in the root
     // frame and there is no storage to propagate.
     frame->storage.eraseAll([](const auto& entry) { return entry.key->isDead(); });
+
     for (auto& entry : frame->storage) {
       storage.insert(entry.clone(js));
     }
   }
 
+  if (storageEntry.key->isDead()) return;
```

Comment:
Oh very unlikely but not impossible. I'd be more concerned about the case someone is using the API directly rather than via AsyncLocalStorage. But even then it's an edge case. I wouldn't consider this a critical change by any stretch

Reply:
I've added a comment pointing out that the case is unlikely but I'd still prefer to keep the additional check, just in case.